purchase_order_analytic_search/models/purchase_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2015 Eficent Business and IT Consulting Services S.L.
+# Copyright 2018 Camptocamp
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    @api.depends("order_line.account_analytic_id")
+    def _compute_analytic_accounts(self):
+        for purchase in self:
+            purchase.account_analytic_ids = purchase.mapped(
+                "order_line.account_analytic_id"
+            )
+
+    @api.model
+    def _search_analytic_accounts(self, operator, value):
+        po_line_obj = self.env["purchase.order.line"]
+        if not value:
+            return [("id", "=", False)]
```

Comment:
Why you return this domain? Maybe you want to get purchase order without account analytic

Reply:
I think the standard behavior when there is not value is not to filter by anything. If filtering by no analytic is needed better to create a specific filter in the view.

Reply:
I see your point.  Would you do that @JuanyDForgeflow ?

Reply:
@AaronHForgeFlow @angelmoya I changed it please review